\documentclass[letterpaper]{article}
\usepackage{natbib,alifeconf}
\usepackage{hyperref}

\title{What is necessary for open-ended evolution?}
\author{Anya E. Vostinar$^{1,2,3}$, Emily L. Dolson$^{1,2,3}$ , Michael J. Wiser$^{1,3}$, \and Charles Ofria$^{1,2,3}$ \\
\mbox{}\\
$^1$BEACON Center for the Study of Evolution in Action  \\
$^2$Department of Computer Science and Engineering, Michigan State University \\
$^3$Program in Ecology, Evolutionary Biology, and Behavior, Michigan State University \\
ofria@msu.edu}


\begin{document}
\maketitle


\section{Introduction}

    A central goal of the field of artificial life is to build systems that demonstrate open-ended evolutionary dynamics, which capture interesting properties of natural systems -- such as the ability to evolve novel cooperative strategies and complex ecosystems -- and undergo major transitions. Such systems are sought after for a number of reasons. For those doing artificial life research, the presence of dynamics that are seen in biology but not artificial life raises the possibility that there is some fundamental and as-of-yet unidentified quality that artificial life systems are missing (citation). Biologists are also interested in these systems because phenomena that would be indicative of open-ended evolution represent some of the most interesting phenomena in biological evolution. Therefore, having an artificial life system capable of producing these behaviors would facilitate understanding them on a mechanistic level (citations). Lastly, insights from open-ended evolutionary systems have the potential to expand the classes of applied engineering problems that we are able to solve with evolutionary algorithms (citations). For instance, biological evolution is currently the only algorithm to have produced general intelligence; harnessing the dynamics that lead to this development could spur breakthroughs in artificial intelligence. While various artificial life systems have recreated individual dynamics -- such as the evolution of complexity, cooperation, and competition (citations) --  the overarching goal is to create a system where all of these dynamics can emerge more organically, as in the biosphere. 
Despite general agreement on open-ended evolution being an important goal in artificial life, the field has yet to come to a consensus on how to define whether systems are making progress towards achieving this goal. A number of dynamics have been proposed as being possibly necessary for open-ended evolution, most notably the continuous production of novelty (cite Stanley and maybe Banzhaf if heâ€™s published that paper yet), the continuous increase in diversity (cite Bedau and maybe others), and the continuous increase in complexity (citations). We argue that all of these dynamics are important pieces of the open-ended evolution puzzle. In addition, we have previously suggested that there is a fourth necessary and even simpler dynamic: continuous change in the population (cite blog post).
These four properties of a system fit into a hierarchy, as shown in Figure 1. For novelty to occur, there must be some degree of change in a population. While this is trivially true, many evolutionary algorithms suffer from premature convergence, which is essentially the absence of change, so it remains an important prerequisite to define. Similarly, complexity and diversity can only increase indefinitely if novel members of the population continue to be generated. All of these dynamics capture different subsets of interesting behavior that an evolving system might have and we propose they are all necessary for a system to demonstrate open-ended evolution.
To start drawing conclusions about what factors of a system promote or inhibit these dynamics, it is critical to have a common suite of metrics that are applicable across a wide variety of systems. Some effort has been made toward this end with evolutionary statistics (cite). Evolutionary statistics focus on the components of a system, which the authors admit will need to be defined for each system as appropriate. Generally, components should capture the individual meaningful pieces of a system, the species in the fossil record being an example. Once these components are decided upon, the diversity and amount of activity of components in a given time step are used to determine which of three classes of evolution the system falls into. These classes of evolutionary activity (none, bounded, unbounded) follow logically from whether diversity and new evolutionary activity per component is bounded or unbounded. However, to measure only meaningful evolutionary activity, a neutral `shadow' control must be run with no selective pressure and the activity from that control subtracted from the activity of the experimental system. We found that the requirement of a shadow control treatment made implementing evolutionary statistics difficult in complex systems; additionally, the need to select an appropriate component and the unintuitive nature of evolutionary statistics has made it difficult to generalize across systems. 
In this paper, we propose to quantify the idea of open-ended evolution with a suite of four necessary but not always sufficient metrics that attempt to balance generalizability across systems with the ability to capture the ideas of the four dynamics previously discussed: change, novelty, complexity, and ecological interactions. We present our results when applying these four metrics to an NK system.


\section{Experimental System}
    To test our metrics, we created an artificial life simulation using an NK model \citep{kauffman_towards_1987}. An NK model uses two parameters, N and K, to randomly generate a fitness landscape. N specifies the number of sites in the genome, each of which is a 0 or a 1. The fitness landscape specifies how much a given value at a given site contributes to the fitness of the bit-string organism. This fitness contribution depends on the values at the K adjacent sites. As such, K tunes the ruggedness of the landscape; low values of K produce smooth landscapes with few peaks, whereas high values produce landscapes with many peaks. We chose to use an NK model because they are a well-understood system for studying general questions about evolutionary dynamics.

\subsection{Measuring Only Persistent Lineages}
    At any given point in time, the population will contain some maladaptive genomes that recently arose via mutation. These genotypes will quickly be purged from the population via natural selection and will add noise to our metrics. To decrease this noise, we must filter out such genomes. We accomplish this by looking backwards in time to see which organisms were the progenitors of lineages that persisted for a substantial number of generations. We mark each organism with a lineage number at a given time point A, as demonstrated in Figure 1 (where color indicates lineage number). The lineage numbers are passed on to offspring throughout the time interval (50 generations in our experiments). At time point A+50, we determine which genomes from the population at A have descendants at A+50. At this point, those genomes are considered persistent; in the example in Figure 1 the green and blue lineages are considered persistent at time point A+50. We then compare the green and blue genomes from time point A to the genomes that were persistent previously, either from time point A-50 or from all previous time points, depending on the metric; in the example we would compare to the purple genome at time point A-50. This filtering leads to a delay in counting a genome in a metric until 50 generations later, but enables us to avoid an apparent increase in metrics due to drift via mutation. For example, the red, orange, and blue genomes from time point A-50 would never be considered in our metrics because their lineages do not persist to time point A.

\begin{figure}
\includegraphics[width=3.5in]{LineageFigure.png}
\caption{\textbf{An illustrative example of how we filter genomes for persistent lineages.} At time point A, the purple lineage has proven to be persistent and therefore the original genome from A-50 will be considered meaningful. Similarly, the green and blue lineages persist to time point A+50 and so the original green and blue genomes will be considered meaningful as they were at time point A.}
\label{lineages}
\end{figure}

\subsection{Measuring Only Informative Sites}
    While a genome may have descendants in 50 generations, it may not be phenotypically different than another persistent genome in the population. To ensure we are not counting phenotypically identical genomes separately, we determine which sites in the genome contain information about the environment. In a an NK bitstring model, the only information that an organism can have about its environment is whether it is a better to have a 1 or a 0 at each site. Thus, informative sites are those for which flipping the corresponding bit would result in a fitness decrease. In calculating all of the following metrics, we first reduce the genome to its informative sites.

\section{Metrics}

\subsection{Change Metric}
    Our first metric focuses on whether the genetic makeup of the population is changing. This metric should always be above zero unless the population has converged and no beneficial variation is being introduced. We use our method of filtering genomes (explained previously) to ensure that we only record a genome as new compared to the previous time point if its lineage has persisted for one full time point. For this comparison, we first find the genomes from persistent lineages from time point A by determining which genomes have descendants in time point A + 50. In the example shown in Figure 1, this would be the genomes at the root of the green and blue lineages. We then compare these genomes to those found to have been from persistent lineages in time point A-50 because they have descendants in time point A, purple in Figure 1. In this way, we create a sliding window to observe change in the population.

\subsection{Novelty Metric}
    The novelty metric measures how many genomes have evolved in the population that have never been seen previously in the experiment. For this metric we again filter out genomes that do not have descendants in the next time point, enabling us to focus on meaningful novelty. To measure novelty, we simply count how many genomes from persistent lineages (genomes that existed in time point A and have descendants in time point A+50) have never been in a previous time pointâ€™s persistent genome pool. It is possible with this metric for a genome to evolve, but not persist, and therefore not be recorded in the permanent history, but then evolve and persist at a later point and be counted as novel. Once a genome has been counted as novel, however, it is part of the permanent history and will never be counted in the novelty metric again. Thus, while a genome could be delayed in being counted as novel, it will not be counted twice.
    

\subsection{Complexity Metric}
    The complexity metric measures if the complexity of the population is increasing. To determine if the complexity is increasing, we only need to focus on the most complex organism in the population at a given time point. We are considering complexity in the information theory sense of the term, and therefore focusing on only informative sites in an organismâ€™s genome. Because the ones of an organismâ€™s genome are the sites that contribute to its fitness, we determine an organismâ€™s complexity by how many ones it has in its genome. Therefore, our complexity measure is simply how many bits are ones in the organism with the most ones in its genome. In our simple simulation, the number of ones is under selection and therefore should increase until all organisms have all ones in their genome. Therefore, the complexity metric cannot increase indefinitely in our simulation, but will demonstrate what can be expected in a closed evolutionary system.

\subsection{Ecological Metric}
    The ecological metric measures the amount of information in the population as a whole. While individuals may not contain increasing amounts of information in single genomes (as measured by the complexity metric), they could still be increasingly diverse and therefore contain increased information collectively in the population. We can measure such a phenomenon by looking at the diversity of persistent genotypes reduced to informative sites. Complex ecologies in which multiple subsets of the population are using different information about the environment to survive are likely to be characterized by a relatively balanced distribution of individuals across the various successful phenotypes. Thus, we use Shannon entropy, a popular metric of diversity that also measures evenness, to measure the diversity of the persistent genotypes and calculate the ecological metric.



\section{Results}
    To ensure that these metrics are capturing the dynamics that we want them to, we tested them on a variety of variants on our basic NK model. Our default configuration was N=20, K=3, population size of 200, mutation rate of 3, with tournament selection and a tournament size of 15.


\subsection{Change and Novelty Metrics}
For a system to exhibit open-ended evolution, it is clearly necessary for it to be changing and introducing novel genomes. Therefore, we first measured the change and novelty of our system to demonstrate what the change and novelty metrics can capture. As shown in Figure 3,



\subsection{Complexity Metric}
    An ideal open-ended evolutionary system would likely have increasingly complex individual organisms. Therefore, we next compared our complexity metric between two of our systems. We calculated the complexity of the most complex organism of each population over time and as Figure 4 shows,


\subsection{Ecological Metric}
Finally, an open-ended evolutionary system should not only have individually complex organisms, but a diverse population of interacting organisms. Even if a system is not evolving more complex individual organisms, it may still be exhibiting interesting dynamics because it allows for a large number of niches and therefore diverse organisms to occupy those niches. Our ecological metric captures this dimension of open-ended evolution. The ecological metric shown in Figure 5 



\section{Conclusions}
Though our simulation is fairly simple, it demonstrates that our metrics respond intuitively to the dynamics in a system. Thus, they should be able to detect systems that are theoretically capable of achieving open-ended evolution. First, it is clear that for new behaviors to evolve in a system, new genomes need to be evolving. Therefore, both the change and novelty metrics would need to be non-zero -- though they could be stable instead of increasing since both metrics measure a change. 
Second, continuously increasing complexity with a stable environment would imply increasing biotic interactions, potentially cooperative or complex competitive interactions. This implication is because if complexity is increasing in a population, at least one organism must be incorporating more information into its genome than any organism had before. This information can be about the environment up to a point. However, if the environment is not changing, complexity will not be able to increase unboundedly without biotic interactions between organisms. Therefore, continuously increasing complexity implies that interesting biotic interactions are evolving in a system, though they could be cooperative or competitive and likely include both.
Finally, a continuously increasing ecological metric in a stable environment implies an ecosystem is forming. This is because if the ecology metric is increasing, the informative diversity in the population is increasing. For diversity to be increasing, new niches must be emerging -- if there was only a single niche, the best genotype for that niche would fill it and take over the population. The only way for diversity to be maintained for a meaningful length of time is for new niches to be created. While new niches can emerge from abiotic factors, if the environment is not changing, those will eventually all fill. Therefore, the only way for an indefinite series of new niches to form (which is required for an unbounded increase of the ecological metric), there must be biotic interactions between organisms in the population. Those biotic interactions could fall into a number of categories such as predator and prey, mutualism, parasitism or commensalism, but as the number of them increase, a complex ecosystem emerges.

 More importantly, we can use these metrics to understand the impact of incremental changes to a system. In order to apply the scientific method to a monolithic problem like designing an open-ended evolutionary system, we need to be able to break that problem down into components that can be addressed in a systematic manner. Moreover, one of the primary goals in building such a system is to understand what components are necessary to do so. By looking at the effects that individual, controlled changes to a system have on this suite of metrics, we can more effectively work towards these goals.


Even biological systems like the LTEE

The metrics we propose here are applicable not only to digital systems, but are also relevant to experimental biological systems.  The Long-Term Evolution Experiment (LTEE) (Lenski et al 1991) is an exemplar of experimental evolution, consisting of 12 populations of the bacteria E. coli which have been evolving independently for more than 60,000 generations.  As detailed in (Taylor et al 2016), the LTEE exhibits many hallmarks of open-ended evolution, including the criteria we propose here.  Because fitness within the LTEE is best described by an unbounded power law function (Wiser et al 2013, Lenski et al 2015), the system meets the change metric; populations continue to change in non-trivial ways over time.  Studies of individual populations within the LTEE have shown numerous examples of generation of novelty, including exploration of new areas of the fitness landscape (Tenaillion et al 2016), repeated selective sweeps (Maddamsetti et al 2015), new diversity arising after such sweeps (Blount et al 2012), and epistasis between late mutations and those which arose earlier (Wielgoss et al 2013), fulfilling the novelty metric.  Several populations within the LTEE demonstrate frequency-dependent fitness dynamics (Ribeck and Lenski 2015, Rozen and Lenski 2000, Le Gac et al 2012, Maddamsetti et al 2015), which are necessarily cases of ecological interactions.  Included in these cases of frequency dependence is a special case (Blount et al 2008, Blunt et al 2012, Turner et al 2015 (PLOS ONE) driven by crossfeeding and specialization on different resources (Turner et al 2015 biorXiv).  Because all of the populations in the LTEE began as single cells, all ecological complexity in any populations must have arisen during the course of the experiment, and thus satisfies the  ecological metric.  The complexity metric is inherently harder to quantify in a biological system than in a computational one, but recent large scale genome sequencing from the LTEE (Tenaillon et al 2016) offers the promise of being able to measure complexity at the genome level over the course of the experiment.  The fact that our metrics work in a well-studied biological example of open-ended evolution lends greater credence to applying them computational studies as well.

References:
Kauffman, S., and Levin, S. (1987). Towards a general theory of adaptive walks on rugged landscapes. Journal of Theoretical Biology 128, 11â€“45.


\bibliographystyle{apalike}
\bibliography{bibliography}

\end{document}
